include/zephyr/net/phy.h

```diff
@@ -99,6 +99,10 @@ typedef void (*phy_callback_t)(const struct device *dev,
  * public documentation.
  */
 __subsystem struct ethphy_driver_api {
+	/** Get MAC address */
+	int (*get_mac_address)(const struct device *dev, uint8_t port,
+			       uint16_t *mac);
```

Comment:
Why `uint16_t`?

Reply:
It was the assumption, that 48-bit of the mac address will be stored in the application and the application will be responsible to deliver the 64-bit buffer, and the driver will increment the 16-bit buffer one time. After your comment it strikes me, that the MAC address can be collected in the driver with appropriate form and delivered in on "shot" by the pointer to buffer `uint64_t *mac` Which seams better isn't?

Reply:
So in general you suggest that the application caller will casts the `uint64_t` address of the buffer to `uint8_t` pointer and pass it to the driver API. And then the API function `get_mac_address()` will be responsible for copying only the 48/64 bits of data. How do you think about @pdgendt solution?